### DISCUSSION TITLE
Reparent to New Node should use its children’s data for XY Position and/or Rect Size

### DISCUSSION THREAD

**Discussion by philippedcote** [godotengine/godot-proposals#4117] [2D]
When using the command "Reparent to New Node", which essentially is a fancy way of saying "Group to New Node", the newly created Node should use its children Rect data for its own Rect data, as it is the case in all industry standard softwares like Figma, Illustrator and others.

This would avoid many, many unnecessary clicks and headaches.

If needed, here are the steps to reproduce using Control Nodes:

1. Add a Control Node.
2. Change its size and position it elsewhere than the default position.
3. Create another Control Node.
4. Change its size and position it elsewhere than the default position and the previous node’s position.
5. Select both Nodes.
6. Use the Reparent to New Node command and choose another Control Node.
7. Notice how the newly created Node completely ignores its children’s Rect data, by using a 40 x 40 Rect Size with a 0, 0 Position.

Notice to reproduce using 2D Nodes: you can try to group 2 sprites inside a Position2D Node.

**Comment by Calinou** [comment]
See the original issue for context: https://github.com/godotengine/godot/issues/58533
